tests/test_workflow/test_preprocess.py

```diff
@@ -315,14 +315,18 @@ def test_create_commands_slf_by_sampleID(self):
 
         all_files = ['sample1_r1.fastq', 'sample2_r1.fastq']
         demultiplexing_method = 'sampleid_by_file'
-        output_dir = "sl_out"
+        # the output directory has to exist
+        output_dir = self.temp_dir
 
         actual_command = create_commands_slf(all_files, demultiplexing_method,
             output_dir)[0][0][1]
 
-        expected_command = "split_libraries_fastq.py  -i sample1_r1.fastq,sample2_r1.fastq --sample_ids sample1,sample2 -o sl_out  --barcode_type 'not-barcoded'"
+        expected = ("split_libraries_fastq.py  -i {output_dir}/seqs.txt -o "
+                    "{output_dir} --read_arguments_from_file --sample_ids "
+                    "{output_dir}/sample_ids.txt --barcode_type "
+                    "'not-barcoded' ").format(output_dir=output_dir)
 
-        self.assertEqual(actual_command, expected_command)
+        self.assertEquals(actual_command, expected)
```

Comment:
`assertEquals` is a [deprecated synonym](https://docs.python.org/2/library/unittest.html#deprecated-aliases), better to use `assertEqual`